tools/build_defs/repo/git.bzl

```diff
@@ -55,7 +58,7 @@ set -ex
 ( cd {working_dir} &&
     if ! ( cd '{dir_link}' && [[ "$(git rev-parse --git-dir)" == '.git' ]] ) >/dev/null 2>&1; then
       rm -rf '{directory}' '{dir_link}'
-      git clone '{shallow}' '{remote}' '{directory}' || git clone '{remote}' '{directory}'
+      git clone '{shallow}' '{single_branch}' '{remote}' '{directory}' || git clone '{remote}' '{directory}'
```

Comment:
Here I'm confused about the quoting. Your `single_branch` variable brings to options (`-b` and `--single-branch`) and should interpolate as 3 `argv` entries to `git`. However, with the given single quotes, those form a single argument.